resources/tentacle.rb

```diff
@@ -19,28 +19,211 @@
-attribute :tenant_tags, kind_of: Array, default: nil
-attribute :tentacle_name, kind_of: String, default: node.name
-attribute :service_user, kind_of: String, default: nil
-attribute :service_password, kind_of: String, default: nil
-attribute :public_dns, kind_of: String, default: node['fqdn']
+action :install do
+  checksum = new_resource.checksum
+  version = new_resource.version
+  upgrades_enabled = new_resource.upgrades_enabled
+
+  verify_version(version)
+  verify_checksum(checksum)
+
+  tentacle_installer = ::File.join(Chef::Config[:file_cache_path], 'octopus-tentacle.msi')
+  install_url = installer_url(new_resource.version)
+
+  remote_file tentacle_installer do
+    action :create
+    source install_url
+    checksum checksum if checksum
+  end
+
+  windows_package display_name do
+    action :install
+    source tentacle_installer
+    version version if version && upgrades_enabled
+    installer_type :msi
+    options installer_options
+  end
+end
+
+action :configure do
+  action_install
+
+  port = resolve_port(polling, new_resource.port)
+  service_name = service_name(instance)
+
+  directory home_path do
+    action :create
+    recursive true
+  end
+
+  powershell_script "generate-tentacle-cert-#{instance}" do
+    action :run
+    cwd tentacle_install_location
+    code <<-EOH
+      .\\Tentacle.exe new-certificate -e "#{cert_file}" --console
+      #{catch_powershell_error('Generating Cert for the machine')}
+    EOH
+    not_if { cert_file.nil? || ::File.exist?(cert_file) }
+  end
+
+  powershell_script "create-instance-#{instance}" do
+    action :run
+    cwd tentacle_install_location
+    code <<-EOH
+      .\\Tentacle.exe create-instance --instance="#{instance}" --config="#{config_path}" --console
+      #{catch_powershell_error('Creating instance')}
+    EOH
+    not_if { ::File.exist?(config_path) }
+  end
+
+  powershell_script "configure-tentacle-#{instance}" do # ~FC009
+    action :run
+    cwd tentacle_install_location
+    sensitive new_resource.sensitive
+    code <<-EOH
+      .\\Tentacle.exe import-certificate --instance="#{instance}" -f #{cert_file} --console
+      #{catch_powershell_error('Importing Certificate that was generated for the machine')}
+      .\\Tentacle.exe new-certificate --instance="#{instance}" --if-blank --console
+      #{catch_powershell_error('Generating Certificate if the Import failed')}
+      .\\Tentacle.exe configure --instance="#{instance}" --reset-trust --console
+      #{catch_powershell_error('Reseting Trust')}
+      .\\Tentacle.exe configure --instance="#{instance}" --home="#{home_path}" --app="#{app_path}" --port="#{port}" --noListen="#{powershell_boolean(polling)}" --console
+      #{catch_powershell_error('Configuring instance')}
+      .\\Tentacle.exe configure --instance="#{instance}" --trust="#{trusted_cert}" --console
+      #{catch_powershell_error('Trusting Octopus Deploy Server')}
+      .\\Tentacle.exe service --instance="#{instance}" --install --console
```

Comment:
Changed this to not start the service.  If we do not notify to restart and the service is already started then the user credentials will not be updated.